lib/winrm/http/transport.rb

```diff
@@ -195,9 +195,11 @@ def seal(message)
         "\x10\x00\x00\x00#{signature}#{emessage}"
       end
 
-      def winrm_decrypt(str)
-        return '' if str.empty?
-        str.force_encoding('BINARY')
+      def winrm_decrypt(resp)
+        # OMI server doesn't always respond to encrypted messages with encrypted responses over SSL
```

Comment:
This is also addressed on the OMI side per my PR at https://github.com/Microsoft/omi/pull/581, but it has not yet released. Presumably it will be a 1.5.1 or similar release.

That PR ensures that if the request was encrypted an encrypted response will be sent (even if over HTTPS).